mtp_api/apps/service/views.py

```diff
@@ -2,25 +2,27 @@
 from django.utils import timezone
 from rest_framework import mixins, viewsets
 
-from service.models import Downtime, SERVICES, Notification
+from service.constants import Service
+from service.models import Downtime, Notification
 from service.serializers import NotificationSerializer
 
 
 def service_availability_view(_):
-    def service_availability(service):
+    def service_availability(service: str):
```

Comment:
I'm not saying change anything as we're already moving a lot of things around but maybe service could be a `Service` here and in `active_downtime()` rather than a String.

I'm just saying this because it took me a few minutes to understand how these methods work and I thought there was a problem but actually this seems fine.

If this was a service everything would be nicer as you're dealing with an higher level concept rather than a string (which could be anything).

e.g.  below you'd have `response = {service: service_availability(service) for service in Service}`

and in `active_downtime()` the query would be more explicit by saying `service`:

```Python
class DowntimeManager(models.Manager):
    def active_downtime(self, service: Service):
        # [...]
        return self.filter(
            # [...]
            service=service.value, # service.value now is explicit
        ).order_by('-end').first()
```

It's just an observation, maybe slightly overkill for this simple methods but probably more meaningful in other places using `TextChoices`.

Reply:
Basically by passing around strings you "loose" information about the enum/TextChoice type (Service in this case). Which could lead to bugs.